tests/unit/preprocessor/RandomFourierGaussPreproc_unittest.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Nanubala Gnana Sai
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/DensePreprocessor.h>
+#include <shogun/preprocessor/RandomFourierGaussPreproc.h>
+
+using namespace shogun;
+
+class RFGPTest : public ::testing::Test
+{
+public:
+	virtual void SetUp()
+	{
+		SGMatrix<float64_t> mat(num_features, num_vectors);
+
+		linalg::range_fill(mat, 1.0);
+
+		auto features = std::make_shared<DenseFeatures<float64_t>>(mat);
+		preproc = std::make_shared<RandomFourierGaussPreproc>();
+		preproc->put("m_dim_output", target_dim);
+		preproc->fit(features);
+	}
+	virtual void TearDown()
+	{
+	}
+
+protected:
+	const int32_t seed = 100;
+	const index_t num_vectors = 5;
+	const index_t num_features = 3;
+	const index_t target_dim = 100;
+	std::shared_ptr<RandomFourierGaussPreproc> preproc;
+};
+
+TEST_F(RFGPTest, approx)
+{
+	SGVector<float64_t> vec_1(num_features);
+	SGVector<float64_t> vec_2(num_features);
+
+	std::mt19937_64 prng(seed);
+
+	NormalDistribution<float64_t> norm_dist;
```

Comment:
no need to put random numbers into the vectors, just range fill them. Or use a list initializer
vec1 = { 1, 2, 3};

Reply:
in light of the comment below, make it a matrix (you can still use a list initialiser)